File: mobile/src/types/workPhoto.ts
export type WorkPhotoItem = {
  photo_id: string
  work_id: string
  farm_cd: string
  sort_no: number
  display_nm: string
  original_nm?: string | null
  stored_nm?: string | null
  file_ext?: string | null
  file_size?: number | null
  width_px?: number | null
  height_px?: number | null
  thumb_url: string
  original_url: string
}

export type WorkPhotoListResponse = {
  work_id: string
  count: number
  max_count: number
  remaining: number
  photos: WorkPhotoItem[]
}

export type WorkPhotoUploadResponse = {
  uploaded: WorkPhotoItem[]
  skipped: string[]
  count: number
  max_count: number
  remaining: number
  message: string
  success: boolean
  photo_id?: string | null
  farm_cd?: string | null
  work_id?: string | null
  error?: string | null
  error_code?: string | null
}

File: server/schemas/work_photo.py
# -*- coding: utf-8 -*-
"""작업 결과 사진 API 스키마."""

from __future__ import annotations

from pydantic import BaseModel, Field


class WorkPhotoItem(BaseModel):
    photo_id: str
    work_id: str
    farm_cd: str
    sort_no: int
    display_nm: str
    original_nm: str | None = None
    stored_nm: str | None = None
    file_ext: str | None = None
    file_size: int | None = None
    width_px: int | None = None
    height_px: int | None = None
    thumb_url: str
    original_url: str


class WorkPhotoListResponse(BaseModel):
    work_id: str
    count: int = Field(ge=0)
    max_count: int = Field(ge=1)
    remaining: int = Field(ge=0)
    photos: list[WorkPhotoItem]


class WorkPhotoUploadResponse(BaseModel):
    uploaded: list[WorkPhotoItem]
    skipped: list[str] = Field(default_factory=list)
    count: int = Field(ge=0)
    max_count: int = Field(ge=1)
    remaining: int = Field(ge=0)
    message: str
    success: bool = True
    photo_id: str | None = None
    farm_cd: str | None = None
    work_id: str | None = None
    file_name: str | None = None
    file_path: str | None = None
    thumbnail_path: str | None = None
    file_size: int | None = None
    width: int | None = None
    height: int | None = None
    created_by: str | None = None
    error: str | None = None
    error_code: str | None = None
